ethcore/src/engines/authority_round/mod.rs

```diff
@@ -545,18 +546,32 @@ impl Engine for AuthorityRound {
 	}
 
 	/// Apply the block reward on finalisation of the block.
-	fn on_close_block(&self, block: &mut ExecutedBlock) -> Result<(), Error> {
+	fn on_close_block(&self, block: &mut ExecutedBlock) -> Result<CloseOutcome, Error> {
+		let tracing_enabled = block.tracing_enabled();
 		let fields = block.fields_mut();
+		let mut tracer = ExecutiveTracer::default();
```

Comment:
nitpick: The tracing happens in `Engine` not `Executive` so we could have a separate struct for tracing the rewards.